### PR TITLE
Update error codes for reporting

### PR DESCRIPTION
**Pull request by gabriel**
Error types for reporting to API.

### CODE REVIEW

**Comment by chrisnojima** [comment]
@keybase/react-hackers

**Comment by chrisnojima** [comment]
logically seems fine to me, looking for some golang review

**Comment by maxtaco** [comment]
@jzila can you help with golang review?

File: error.go [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+	source    error
+}
+
+// NewUpdateError constructs an UpdateError from a source error
+func NewUpdateError(errorType UpdateErrorType, err error) error {
+	return UpdateError{errorType: errorType, source: err}
+}
+
+// TypeString returns a unique short string to denote the error type
+func (e UpdateError) TypeString() string {
```
Comment (jzila) [inline error.go]: this probably belongs in `UpdateErrorType`:
```go
func (t UpdateErrorType) String() string {
    return string(t)
}
```

**Comment by jzila** [comment]
LGTM except for nit

**Pull request merged**